src/test/HostActivationTests/DotNetBuilder.cs

```diff
@@ -20,6 +21,18 @@ public class DotNetBuilder
         private readonly string _path;
         private readonly RepoDirectoriesProvider _repoDirectories;
 
+        public enum MockType
+        {
+            HostPolicy,
+            CoreClr,
+        };
+
+        private static readonly Dictionary<MockType, List<ValueTuple<string, string>>> _mockTypesDict = new Dictionary<MockType, List<ValueTuple<string, string>>>()
+        {
+            { MockType.HostPolicy, new List<ValueTuple<string, string>>(){ ("hostpolicy","mockhostpolicy") }},
+            { MockType.CoreClr,    new List<ValueTuple<string, string>>(){ ("hostpolicy","hostpolicy"), ("coreclr","mockcoreclr") }},
```

Comment:
I would prefer separate methods to handle these.
For one the `hostpolicy` in the `CoreClr` case won't work (since we need to take the product built `hostpolicy` which will not be `corehost_test`.
For another, we will need to add `.deps.json` file to make the framework work with the real `hostpolicy`.